modules/migration-migrating-applications-api.adoc

```diff
@@ -0,0 +1,393 @@
+----
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  name: host <1>
+  namespace: openshift-migration
+spec:
+  isHostCluster: true <2>
+----
+<1> `host` is the default name of the cluster on which {mtc-short} is installed. You can change the cluster name.
+<2> If the `migration-controller` CR is not running on the target cluster, change `isHostCluster` to `false`.
+
+. Create the `MigCluster` CR for the target cluster:
++
+[source,terminal]
+----
+$ oc create -f target-cluster.yaml -n openshift-migration
+----
+
+. Create a `Secret` manifest for the source cluster called `source-secret.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <source-cluster-secret>
+  namespace: openshift-config
+type: Opaque
+data:
+  saToken: <sa_token> <1>
+----
+<1> Specify the base64-encoded `migration-controller` service account (SA) token of the source cluster.
++
+You can obtain the SA token by running the following command:
++
+[source,terminal]
+----
+$ oc sa get-token migration-controller -n openshift-migration | base64 -w 0
+----
+
+. Create the `Secret` CR for the source cluster:
++
+[source,terminal]
+----
+$ oc create -f source-secret.yaml
+----
+
+. Create a `MigCluster` manifest for the source cluster called `source-cluster.yaml`:
++
+[source,yaml]
+----
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  name: <source_cluster>
+  namespace: openshift-migration
+spec:
+# Uncomment for direct image migration
+# exposedRegistryPath: <exposed_registry_route> <1>
+  insecure: false <2>
+  isHostCluster: false
+  serviceAccountSecretRef:
+    name: <source_cluster_secret> <3>
+    namespace: openshift-config
+  url: <source_cluster_url> <4>
+----
+<1> If you are performing direct image migration, specify the exposed registry route, for example, `docker-registry-default.apps.mycluster-ocp3.com`.
+<2> `insecure`: SSL verification is enabled if `false`. CA certificates are not required or checked if `true`.
+<3> Specify the `Secret` CR of the source cluster.
+<4> Specify the URL of the source cluster.
+
+. Create the `MigCluster` CR for the source cluster:
++
+[source,terminal]
+----
+$ oc create -f source-cluster.yaml -n openshift-migration
+----
+
+. Verify that the source and target clusters are in a `Ready` state:
++
+[source,terminal]
+----
+$ oc describe cluster <cluster_name>
+----
+
+. Create a `Secret` manifest for the replication repository called `storage-secret.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: openshift-config
+  name: <migstorage_creds>
+type: Opaque
+data:
+  aws-access-key-id: <key_id_base64> <1>
+  aws-secret-access-key: <secret_key_base64> <2>
+----
+<1> Specify the key ID in base64 format.
+<2> Specify the secret key in base64 format.
++
+AWS credentials are base64-encoded by default. If you are using another storage provider, you must encode your credentials by running the following command with each key:
```

Comment:
Can you provide the missing information for Azure and GCP?

Reply:
This is a jinja2 templte of a Secret that provides the Azure necessary credentials:

```
{% set credentials_value = 'AZURE_SUBSCRIPTION_ID=' + az_subscription_id  + '\n' +
                           'AZURE_TENANT_ID=' + az_tenant_id  + '\n' +
                           'AZURE_CLIENT_ID=' + az_client_id  + '\n' +
                           'AZURE_CLIENT_SECRET=' + az_client_secret
 %}

apiVersion: v1
kind: Secret
metadata:
  namespace: {{ migration_namespace }}
  name: {{ migstorage_name }}-creds
type: Opaque
data:
  azure-credentials: {{  credentials_value | string | b64encode }}
```

Reply:
The right way to create an azure migstorage is described in the documentation here too https://deploy-preview-29058--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/configuring-replication-repository-3-4.html#migration-configuring-azure_migrating-3-4

Reply:
@sergiordlr  Since we already describe how to configure the Azure storage, including the credentials, I don't think we need to add it to this section. However, it might be a good idea to say that the object storage must be **configured** for MTC in the "Prerequisites" section. That means they will have done the AWS, Azure, GCP, configuration and they will have created the various keys and credentials.